src/iris/api.py

```diff
@@ -700,6 +708,26 @@ def process_resource(self, req, resp, resource, params):  # pragma: no cover
         if resource.allow_read_no_auth and method == 'GET':
             return
 
+        # Ignore HMAC requirements for custom webhooks
+        if req.env['PATH_INFO'].startswith('/v0/webhooks/'):
+            app_name = req.get_param('application', required=True)
+            app = cache.applications.get(app_name)
+            if not app:
+                logger.warn('Tried authenticating with nonexistent app: "%s"', app_name)
+                raise HTTPUnauthorized('Authentication failure', '', [])
+
+            req.context['app'] = app
+
+            # determine if we're correctly using an application key
+            api_key = req.get_param('key', required=True)
+            if not api_key:
+                logger.warn('Did not provide application key for "%s"', app_name)
+                raise HTTPUnauthorized('Authentication failure', '', [])
+            if not api_key == str(app['key']):
```

Comment:
should probably be `if not equals(api_key, str(app['key']))` to prevent memory timing attacks. (`equals` is already imported and used elsewhere in the auth checking)